// @react-aria/ssr/dist/module.js
import $fZ9Rb$react, {useContext as $fZ9Rb$useContext, useMemo as $fZ9Rb$useMemo, useState as $fZ9Rb$useState, useLayoutEffect as $fZ9Rb$useLayoutEffect} from "react";

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $6f523efd7517f7a2$exports = {};

$parcel$export($6f523efd7517f7a2$exports, "SSRProvider", () => $6f523efd7517f7a2$export$9f8ac96af4b1b2ae);
$parcel$export($6f523efd7517f7a2$exports, "useSSRSafeId", () => $6f523efd7517f7a2$export$619500959fc48b26);
$parcel$export($6f523efd7517f7a2$exports, "useIsSSR", () => $6f523efd7517f7a2$export$535bd6ca7f90a273);

// Default context value to use in case there is no SSRProvider. This is fine for
// client-only apps. In order to support multiple copies of React Aria potentially
// being on the page at once, the prefix is set to a random number. SSRProvider
// will reset this to zero for consistency between server and client, so in the
// SSR case multiple copies of React Aria is not supported.
const $6f523efd7517f7a2$var$defaultContext = {
    prefix: String(Math.round(Math.random() * 10000000000)),
    current: 0
};
const $6f523efd7517f7a2$var$SSRContext = /*#__PURE__*/ $fZ9Rb$react.createContext($6f523efd7517f7a2$var$defaultContext);
function $6f523efd7517f7a2$export$9f8ac96af4b1b2ae(props) {
    let cur = $fZ9Rb$useContext($6f523efd7517f7a2$var$SSRContext);
    let value = $fZ9Rb$useMemo(()=>({
            // If this is the first SSRProvider, start with an empty string prefix, otherwise
            // append and increment the counter.
            prefix: cur === $6f523efd7517f7a2$var$defaultContext ? '' : `${cur.prefix}-${++cur.current}`,
            current: 0
        })
    , [
        cur
    ]);
    return(/*#__PURE__*/ $fZ9Rb$react.createElement($6f523efd7517f7a2$var$SSRContext.Provider, {
        value: value
    }, props.children));
}
let $6f523efd7517f7a2$var$canUseDOM = Boolean(typeof window !== 'undefined' && window.document && window.document.createElement);
function $6f523efd7517f7a2$export$619500959fc48b26(defaultId) {
    let ctx = $fZ9Rb$useContext($6f523efd7517f7a2$var$SSRContext);
    // If we are rendering in a non-DOM environment, and there's no SSRProvider,
    // provide a warning to hint to the developer to add one.
    if (ctx === $6f523efd7517f7a2$var$defaultContext && !$6f523efd7517f7a2$var$canUseDOM) console.warn('When server rendering, you must wrap your application in an <SSRProvider> to ensure consistent ids are generated between the client and server.');
    return $fZ9Rb$useMemo(()=>defaultId || `react-aria${ctx.prefix}-${++ctx.current}`
    , [
        defaultId
    ]);
}
function $6f523efd7517f7a2$export$535bd6ca7f90a273() {
    let cur = $fZ9Rb$useContext($6f523efd7517f7a2$var$SSRContext);
    let isInSSRContext = cur !== $6f523efd7517f7a2$var$defaultContext;
    let [isSSR, setIsSSR] = $fZ9Rb$useState(isInSSRContext);
    // If on the client, and the component was initially server rendered,
    // then schedule a layout effect to update the component after hydration.
    if (typeof window !== 'undefined' && isInSSRContext) // This if statement technically breaks the rules of hooks, but is safe
    // because the condition never changes after mounting.
    // eslint-disable-next-line react-hooks/rules-of-hooks
    $fZ9Rb$useLayoutEffect(()=>{
        setIsSSR(false);
    }, []);
    return isSSR;
}




export {$6f523efd7517f7a2$export$9f8ac96af4b1b2ae as SSRProvider, $6f523efd7517f7a2$export$619500959fc48b26 as useSSRSafeId, $6f523efd7517f7a2$export$535bd6ca7f90a273 as useIsSSR};
